src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -279,6 +204,50 @@ const EditShiftHours = ({
     );
   };
 
+  const hasShiftMovedFromTimecard = (startDate, endDate) => {
+    const startDayTime = dayjs(timecardDate).startOf('day');
+    const endDayTime = dayjs(timecardDate).endOf('day');
+    const currentDayStart = new Date(startDayTime).getTime();
+    const currentDayEnd = new Date(endDayTime).getTime();
+    const startDay = new Date(startDate).getTime();
+    const endDay = new Date(endDate).getTime();
+
+    const singleDateMoved =
+      !finishEntryExists &&
+      (startDay < currentDayStart || startDay > currentDayEnd);
+
+    const bothDatesMoved =
+      finishEntryExists &&
+      (startDay > currentDayEnd || endDay < currentDayStart);
+
+    if (singleDateMoved || bothDatesMoved) {
+      hasShiftMovedCallback();
+      return true;
+    }
+    return false;
+  };
+
+  const setMessages = (startDate, endDate) => {
+    const formattedStart = formatDate(startDate);
+    const formattedEnd = formatDate(endDate);
+
+    if (!finishEntryExists) {
+      summaryMessages['update'] = {
+        template: `singleDateMoved`,
```

Comment:
If we go with the composition related comment, instead of passing in a template to be associated with a component name, we could just pass a boolean for `singleDateMoved`, then in the `MessageItem` if `singleDateMoved` is false we can additionally render the `DoubleDateMoved` stuff

Reply:
Resolving after discussion based on other related comment.